bazel/repository_locations.bzl

```diff
@@ -15,10 +15,10 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Gazelle",
         project_desc = "Bazel BUILD file generator for Go projects",
         project_url = "https://github.com/bazelbuild/bazel-gazelle",
-        version = "0.22.2",
-        sha256 = "b85f48fa105c4403326e9525ad2b2cc437babaa6e15a3fc0b1dbab0ab064bc7c",
+        version = "0.22.3",
```

Comment:
Can we bump this dependency to the latest version `0.24.0`?

https://github.com/bazelbuild/bazel-gazelle/releases/tag/v0.24.0

Reply:
As this is a stable branch I'd like to keep the changes minimal. 1.21 and main already have this latest version.

Reply:
`0.22.3` was released in December 2020, `0.23.0` in March 2021 and `0.24.0` in October 2021. As a @envoyproxy/dependency-shepherds I'd rather bump to the latest version which is what we typically strive for. I've been running `0.24.0` in my local environment on Linux aarch64 and amd64 and MacOS aarch64 and amd64 without issue.

Reply:
ah ok, got it. Since the impact is very low, I made the changes. Thanks for explaining.